Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzToolsFramework/API/EntityCompositionNotificationBus.h>
+#include <AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h>
+#include <AzToolsFramework/ComponentMode/EditorComponentModeBus.h>
+#include <AzToolsFramework/Entity/EntityTypes.h>
+#include <AzToolsFramework/ViewportUi/Button.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiRequestBus.h>
+
+namespace AZ
+{
+    class Component;
+}
+namespace AzToolsFramework
+{
+    namespace ComponentModeFramework
+    {
+
+        enum class AddOrRemoveComponent : AZ::u8
+        {
+            Add,
+            Remove
+        };
+
+        // Struct containing relevant information about component for the switcher
```

Comment:
nit: This comment should be in the `//!` style (see https://github.com/o3de/sig-core/blob/main/governance/API-Ref-Guidelines-Update.md for more details)